distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso

```diff
@@ -154,20 +155,95 @@ list_directory directory name_filter=Nothing recursive=False =
    - headers: The headers to send with the request. Defaults to an empty vector.
    - try_auto_parse:  If successful should the body be attempted to be parsed to
      an Enso native object.
+
+   > Example
+     Read from an HTTP endpoint.
+
+         import Standard.Base.Data
+         response = Data.fetch URL
+
+   > Example
+     Read from an HTTP endpoint and write the results to a file.
+
+         import Standard.Base.Data
+         file = enso_project.data / "spreadsheet.xls"
+         HTTP.fetch URL . to_file file
 fetch : (URI | Text) -> HTTP_Method -> Vector (Header | Pair Text Text) -> Boolean -> Any
-fetch uri method=HTTP_Method.Get headers=[] try_auto_parse=True =
-    parsed_headers = headers . map h-> case h of
-        _ : Vector -> Header.new (h.at 0) (h.at 1)
-        _ : Pair -> Header.new (h.at 0) (h.at 1)
-        _ : Header -> h
-        _ -> Error.throw (Illegal_Argument.Error "Invalid header type - all values must be Vector, Pair or Header (got "+(Meta.get_simple_type_name h)+").")
-
-    request = Request.new method uri parsed_headers
-    response = HTTP.new.request request
-
-    if response.code.is_success.not then Error.throw (Request_Error.Error "Status Code" ("Request failed with status code: " + response.code.to_text + ". " + response.body.decode_as_text)) else
-        if try_auto_parse.not then response else
-            response.decode if_unsupported=response . catch handler=(_->response)
+fetch (uri:(URI | Text)) (method:HTTP_Method=HTTP_Method.Post) (headers:(Vector (Header | Pair Text Text))=[]) (try_auto_parse_response:Boolean=True) =
+    HTTP.fetch uri method headers try_auto_parse_response
+
+## ALIAS upload, http post
+   GROUP Input
+   Writes the provided data to the provided URI. Returns the response,
+   parsing the body if the content-type is recognised. Returns an  error if the
+   status code does not represent a successful response.
+
+   Arguments:
+   - uri: The URI to fetch.
+   - body: The data to write.
+   - method: The HTTP method to use. Defaults to `GET`.
```

Comment:
```suggestion
   - method: The HTTP method to use. Defaults to `POST`.
```

The default is set correctly, just the comment needs updating.

Reply:
Done